2-mandatory/3-shopping-list.js

```diff
@@ -19,7 +19,23 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+        //create an array of all the ingredients from the pantry
+        let availableIngredients = Object.values(pantry).flat()//["butter", "milk", "salt", "tinned tomatoes", "oregano"]
+        
+
+  let missingItems = []
+  let shoppingListObject = {}
+  for (let item of recipe["ingredients"]) {
+
+    if (!availableIngredients.includes(item)) {
+      missingItems.push(item)
+    }
+  
+    shoppingListObject["name"] = recipe.name;
+    shoppingListObject["items"] = missingItems;
```

Comment:
(stretch challenge)  While this works, each time the for loop is running, the shoppingListObject is being assigned to the new value of missingItems and name is being reassigned to be the same name, can you see what issue might come from this if the function started being used for really really large inputs? and can you see how you could fix it?